src/lib/common/sol-platform-impl-linux-micro.c

```diff
@@ -514,16 +521,108 @@ teardown_pid1(void)
     reboot(reboot_cmd);
 }
 
+static bool
+sol_platform_linux_micro_should_debug(struct gdbserver *gdb)
+{
+    int idx, argc = sol_argc();
+    char **argv = sol_argv();
+    bool res = false;
+
+    for (idx = 1; idx < argc; idx++) {
+        if (streq(argv[idx], SOL_DEBUG_ARG))
+            res = true;
+        else if (strstartswith(argv[idx], SOL_DEBUG_COMM_ARG)) {
+            gdb->comm = argv[idx] + strlen(SOL_DEBUG_COMM_ARG);
+            if (res) {
+                SOL_DBG("Got "SOL_DEBUG_ARG" flag and comm");
+                return res;
+            }
+        }
+    }
+
+    return res;
+}
+
+static void
+on_fork(void *data)
+{
+    int err;
+    unsigned int i;
+    char pids[PATH_MAX];
+    const char *gdbserver = NULL;
+    struct gdbserver *gdb = data;
+
+    const char *paths[] = {
+        "/usr/bin/gdbserver",
+        "/bin/gdbserver",
+    };
+
+    for (i = 0; i < ARRAY_SIZE(paths); i++) {
+        struct stat st;
+        err = stat(paths[i], &st);
+        if (err < 0) {
+            if (errno == ENOENT) {
+                SOL_DBG("Could not find gdbserver executable at: %s", paths[i]);
+                continue;
+            } else {
+                SOL_DBG("Could not stat gdbserver executable at: %s - %s",
+                        paths[i], sol_util_strerrora(errno));
+                continue;
+            }
+        } else if (!(st.st_mode & S_IXUSR)) {
+            SOL_DBG("gdbserver executable at: %s - is not executable", paths[i]);
+            sol_platform_linux_fork_run_exit(EXIT_FAILURE);
+            continue;
+        }
+
+        gdbserver = paths[i];
+        break;
+    }
```

Comment:
coverity will bitch about  this `stat()` loop, at least it did for other cases like that. I guess we can try to exec them directly, if fail, then we try the next one.